main.js

```diff
@@ -0,0 +1,17 @@
+// - [ ] Remove the "All Posts" link from the header.
+// - [ ] Remove the `span` element that contains the "Read Time" in the featured post. The featured post is the first post on the page.
+// - [ ] Remove the last post from the page, titled "Stop Planning".
+// - [ ] Remove all titles from all non-featured posts.
+
+let linkInHeader = document.querySelector("header nav ul li");
+let spanElem = document.querySelector(".featured span");
+let lastPost = document.querySelector("main .posts article");
+let titles = document.querySelectorAll("main .posts h3")
+let titlesArr = Array.from(titles)
+
+linkInHeader.remove();
+spanElem.remove();
+lastPost.remove();
+for(let title of titlesArr){
+    title.remove()
+}
```

Comment:
```suggestion
let linkInHeader = document.querySelector("header nav ul li");
let spanElem = document.querySelector(".featured span");
let allPosts = document.querySelectorAll("main .posts article");
let lastPost = allPosts[allPosts.length - 1]
let titles = document.querySelectorAll("main .posts h3")

linkInHeader.remove();
spanElem.remove();
lastPost.remove();
for(let title of titles){
    title.remove()
}
```
What you originally had saved as `lastPost` is actually the **first** post. `querySelector` grabs the **first** matching element. In order to grab the **last** matching element out of several matching elements, you have to use `querySelectorAll`. `querySelectorAll` returns a **node list** which is very similar to an array. You can grab the last element from the node list in the same way that you would grab the last element of an array: `allPosts[allPosts.length - 1]`

I got rid of line 10 `let titlesArr = Array.from(titles)` because it's not necessary to convert your `titles` node list into an array if you're just iterating over it with a normal for... of loop. The only time you would need to convert a node list to an array is if you wanted to use array methods on it: .filter, .map, .some, .every, .find, etc. (.forEach actually **_does_** work on node lists, but that is the one exception.)